sdk/python/kfp/v2/compiler/compiler.py

```diff
@@ -871,6 +905,11 @@ def _group_to_dag_spec(
           deployment_config.executors[
               executor_label].custom_job.custom_job.update(custom_job_spec)
 
+      # populates metrics outputs
+      self._populate_metrics_in_dag_outputs(subgroup_component_spec,
+                                            subgroup_task_spec,
+                                            group_component_spec)
```

Comment:
This currently won't work in 2+ layers of dag. But that only happens when there's condition or loop. We can use this to test simple flatten pipeline at least.

Reply:
/hold

While the compiled lightweight function sample  is still ready for testing, I'll make some fix so that it works on multi level dag.

Reply:
/unhold

Fixed the aforementioned issue, this is now ready for review.